burn-tensor/src/tensor/api/base.rs

```diff
@@ -168,6 +173,217 @@ where
     }
 }
 
+impl<B, const D: usize> Tensor<B, D, Int>
+where
+    B: Backend,
+{
```

Comment:
I think you should make the function generic over the kind with `K: BasicOps`. The same code can be used for all tensor kinds.

Reply:
Ah that's how you do it. I thought there was a better way than defining an `impl ` for everything. Thanks!